holoviews/plotting/plotly/selection.py

```diff
@@ -20,7 +20,45 @@ def _build_element_layer(
         else:
             shared_opts = dict()
 
-        merged_opts = dict(self._get_color_kwarg(layer_color), **shared_opts)
-        layer_element = element.options(visible=visible, **merged_opts)
+        merged_opts = dict(shared_opts)
+
+        if layer_color is not None:
+            # set color
+            merged_opts.update(self._get_color_kwarg(layer_color))
+        else:
+            # Keep current color (including color from cycle)
+            for color_prop in self.color_props:
+                current_color = element.opts.get(group="style")[0].get(color_prop, None)
+                if current_color:
+                    merged_opts.update({color_prop: current_color})
+
+        layer_element = element.options(**merged_opts)
 
         return layer_element
+
+    def _style_region_element(self, region_element, unselected_color):
+        from ..util import linear_gradient
+        backend_options = Store.options(backend="plotly")
+        element_name = type(region_element).name
+        style_options = backend_options[(type(region_element).name,)]['style']
+        allowed_keywords = style_options.allowed_keywords
+        options = {}
+
+        if element_name != "Histogram":
+            # Darken unselected color
+            region_color = linear_gradient(unselected_color, "#000000", 9)[3]
+            if "line_width" in allowed_keywords:
+                options["line_width"] = 1
+        else:
+            # Darken unselected color slightly
+            region_color = linear_gradient(unselected_color, "#000000", 9)[1]
```

Comment:
And also adjust alpha here?  Probably should pull these out into a function (maybe one that can be replaced as a parameter) for computing the new color given some arguments.